### PR TITLE
Added a method rowAdd(...) to Matrix 

### PR DESCRIPTION
**Pull request by christianbender**
Issue #64 
I added the feature as requested:  

> Adds a multiple of one row to another inside a matrix.

```js
rowAdd(dest : number, source : number, scalar : number = 1.0 ) : Matrix
```

Inclusive tests

### CODE REVIEW

**Comment by mateogianolio** [comment]
Thanks, will have a look at it this weekend.

**Comment by christianbender** [comment]
@christianbender Super, Thanks.

File: src/Matrix.prototype.spec.ts [hunk 1]
```diff
@@ -343,4 +343,19 @@ describe('Matrix.prototype', () => {
...
+      let m = new Matrix([[1, 2], [3, 4]]);
+      m.rowAdd(0, 1, 10);
+      assert.deepEqual([[31, 42], [3, 4]], m.toArray());
+    });
+
+    it('should throw an exception', () => {
+      let m = new Matrix([[1, 2], [3, 4]]);
+      assert.throws(m.rowAdd.bind(3, 1, 10), Error);
+      assert.throws(m.rowAdd.bind(0, 5, 10), Error);
+      assert.throws(m.rowAdd.bind(0, 1, 'a'), Error);
```
Comment (mateogianolio) [inline src/Matrix.prototype.spec.ts]: This line causes the build to fail. Remove it.

File: src/Matrix.ts [hunk 2]
```diff
@@ -801,6 +801,41 @@ export default class Matrix extends NDArray {
...
+   *          the scalar (for adding)
+   *        - scalar : the scalar of type number.
+   * OUTPUT: The current changed matrix.
+   */
+  rowAdd(dest : number, source : number, scalar : number = 1.0 ) : Matrix {
+
+    // preconditions
+    this.check(dest, 0);
+    this.check(source, 0);
+    if (typeof scalar != "number")
```
Comment (mateogianolio) [inline src/Matrix.ts]: There are no runtime type checks in the other methods, no reason to do it here.

**Comment by christianbender** [comment]
@mateogianolio 
* I changed the method ```rowAdd(...)``` as requested and made it more effiecient. 
* In addition I removed the runtime type checks and changed the test-suite accordingly. Removed the exception-test

File: src/Matrix.ts (hunk 2 above)
Comment (mateogianolio) [reply src/Matrix.ts]: @christianbender Sorry, I only meant the `if (typeof scalar != "number")`! The out of bounds checks (`this.check(dest, 0)` and `this.check(source, 0)`) should definitely still be there :)

**Comment by christianbender** [comment]
@mateogianolio Done. I added the preconditions again

**Pull request merged**